### ISSUE TITLE
Using RMM pool

### ISSUE THREAD

**Issue by jakirkham** [benjha/nvrapids_olcf#3]
When starting `dask-cuda-worker`, would recommend passing [`--rmm-pool-size`]( https://github.com/rapidsai/dask-cuda/blob/8df4f1ca0fb76ec3598717163bec5e0f99705d5d/dask_cuda/cli/dask_cuda_worker.py#L95-L101 ) with an amount of memory equal to the memory on a single device. For example, if one has 4 GPUs with 8GB a piece, this should be `--rmm-pool-size 7GB`.

cc @pentschev (in case you have more thoughts here)

**Comment by benjha** [comment]
Thanks John, 

We'll be updating the repo to Rapids 0.14

So, this option is to get some performance improvements when several  allocations/deallocations are happening, right ?

Is there any particular allocator (a.k.a. device_memory_resource) being used if the flag is not set ?

**Comment by jakirkham** [comment]
Yeah that's the idea. This will route memory allocations through RMM's memory pool (currently [CNMeM]( https://github.com/NVIDIA/cnmem ) though that's an implementation detail).

By default it just goes through `cudaMalloc`, which will be pretty slow.

**Comment by pentschev** [comment]
You should not try to use a pool of 100% of the GPU size though, as there are things like the CUDA context that will consume some memory too. Usually we should leave 500MB-1GB out of the pool, so for an 8GB GPU, we would probably use only 7000MB or 7500MB.

**Comment by jakirkham** [comment]
Good point Peter! Libraries can take up GPU memory too. Updated the recommendation in the OP accordingly.
